### PR TITLE
arm/dts: Add overlay for RevPi HAT EEPROMs [REVPI-2711]

### PR DESCRIPTION
**Pull request by iluminat23**
This overlay can be used to make the RevPi HAT EEPROM available. The overlay should only be used to flash the HAT EEPROM. Loading this overlay by default might have undesired side effects. The HAT EEPROM is attached to the i2c0 which is shared with the video core. On some RevPi devices it might trigger a reset of the periphery if the reset is not actively disabled. How the reset is disable is board specific.

Signed-off-by: Philipp Rosenberger <p.rosenberger@kunbus.com>

### CODE REVIEW

File: arch/arm/boot/dts/overlays/revpi-hat-eeprom-overlay.dts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ */
+
+/dts-v1/;
+/plugin/;
+
+/{
+	compatible = "brcm,bcm2835";
+
+	fragment@0 {
+		target = <&i2c0if>;
```
Comment (linosanfilippo-kunbus) [inline arch/arm/boot/dts/overlays/revpi-hat-eeprom-overlay.dts:13]: Typo: i2c0if -> i2c0

File: arch/arm/boot/dts/overlays/revpi-hat-eeprom-overlay.dts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+			status = "okay";
+
+			eeprom: eeprom@50 {
+				compatible = "atmel,24c256";
+				reg = <0x50>;
+				pagesize = <64>;
+				status = "okay";
+			};
+		};
+	};
```
Comment (linosanfilippo-kunbus) [inline arch/arm/boot/dts/overlays/revpi-hat-eeprom-overlay.dts:29]: In our other overlays we leave one blank before a new fragment. We should do this here, too, to be consistent.

File: arch/arm/boot/dts/overlays/revpi-hat-eeprom-overlay.dts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	fragment@1 {
+		target-path = "/aliases";
+		__overlay__ {
+			i2c0 = "/soc/i2c@7e205000";
+		};
+	};
+	fragment@8 {
```
Comment (linosanfilippo-kunbus) [inline arch/arm/boot/dts/overlays/revpi-hat-eeprom-overlay.dts]: why not use fragment@2 here?

**Review by linosanfilippo-kunbus — CHANGES_REQUESTED** [review]

File: arch/arm/boot/dts/overlays/revpi-hat-eeprom-overlay.dts (hunk 2 above)
Comment (iluminat23) [reply arch/arm/boot/dts/overlays/revpi-hat-eeprom-overlay.dts:29]: right!

File: arch/arm/boot/dts/overlays/revpi-hat-eeprom-overlay.dts (hunk 3 above)
Comment (iluminat23) [reply arch/arm/boot/dts/overlays/revpi-hat-eeprom-overlay.dts]: Coyp-paste error

File: arch/arm/boot/dts/overlays/revpi-hat-eeprom-overlay.dts (hunk 1 above)
Comment (iluminat23) [reply arch/arm/boot/dts/overlays/revpi-hat-eeprom-overlay.dts:13]: No i2c0if is acctually correct. As the i2c0 is shared with the video core, this node is rather strange.
There is the i2c0if node:
https://github.com/RevolutionPi/linux/blob/a80991905191b94e5f5128d45e17a2fec700c6af/arch/arm/boot/dts/bcm283x.dtsi#L339
And the i2c0mux node:
https://github.com/RevolutionPi/linux/blob/a80991905191b94e5f5128d45e17a2fec700c6af/arch/arm/boot/dts/bcm283x.dtsi#L349
The i2c0mux node has a i2c0 node:
https://github.com/RevolutionPi/linux/blob/a80991905191b94e5f5128d45e17a2fec700c6af/arch/arm/boot/dts/bcm283x.dtsi#L360
But the i2c0 node for the i2cmux node is only needed when the i2c is shared with the camera interface. At least this is how I understand this.

And that's the reason for the `alias` and the `__symbols__` fragment.

TL;DR:
I used the i2c0 overlay as base for this overlay:
https://github.com/RevolutionPi/linux/blob/revpi-5.10/arch/arm/boot/dts/overlays/i2c0-overlay.dts

Comment (linosanfilippo-kunbus) [reply arch/arm/boot/dts/overlays/revpi-hat-eeprom-overlay.dts:13]: Interesting, thx for the explanation!

**Review by linosanfilippo-kunbus — DISMISSED** [review]

**Comment by linosanfilippo-kunbus** [comment]
There is a small typo in the last sentence of the commit message: How the reset is disable ... -> How the reset is disabled. But this is rather a minor issue so I  approved this nevertheless (and otherwise it looks good to me). Its up to you if you want to fix that typo.

**Review by linosanfilippo-kunbus — APPROVED** [review]
